apps/client-server/src/app/websites/implementations/bluesky/bluesky.website.ts

```diff
@@ -0,0 +1,698 @@
+import { WebsiteMetadata } from '../../decorators/website-metadata.decorator';
+import { DataPropertyAccessibility } from '../../models/data-property-accessibility';
+import { FileWebsite } from '../../models/website-modifiers/file-website';
+import { MessageWebsite } from '../../models/website-modifiers/message-website';
+import { OAuthWebsite } from '../../models/website-modifiers/oauth-website';
+import { Website } from '../../website';
+import { BlueskyFileSubmission } from './models/bluesky-file-submission';
+import { BlueskyMessageSubmission } from './models/bluesky-message-submission';
+
+@WebsiteMetadata({
+  name: 'bluesky',
+  displayName: 'BlueSky',
+})
+@CustomLoginFlow()
+@SupportsUsernameShortcut({
+  id: 'bsky',
+  url: 'https://bsly.app/profile/$1',
+})
+@SupportsFiles({
+  acceptedMimeTypes: [
+    'image/png',
+    'image/jpeg',
+    'image/gif',
+    'video/mp4',
+    'video/mov',
+    'video/webm',
+  ],
+  acceptedFileSizes: { '*': 1_000_000 },
+  fileBatchSize: 4,
+})
+export default class Bluesky
+  extends Website<BlueskyAccountData>
+  implements
+    FileWebsite<BlueskyFileSubmission>,
+    MessageWebsite<BlueskyMessageSubmission>,
+    OAuthWebsite<BlueskyOAuthRoutes>
+{
+  onAuthRoute: OAuthRouteHandlers<BlueskyOAuthRoutes> = {
+    login: async (request) => {
+      await this.setWebsiteData(request);
+      const result = await this.onLogin();
+      return { result: result.isLoggedIn };
+    },
+  };
+
+  protected BASE_URL = 'https://bsky.app/';
+
+  readonly MAX_CHARS = 300;
+
+  public externallyAccessibleWebsiteDataProperties: DataPropertyAccessibility<BlueskyAccountData> =
+    {
+      username: true,
+      password: true,
+    };
+
+  private createAgent(): AtpAgent {
+    return new AtpAgent({ service: 'https://bsky.social' });
+  }
+
+  public async onLogin(): Promise<ILoginState> {
+    const { username, password } = this.websiteDataStore.getData();
+
+    const agent = this.createAgent();
+    return agent
+      .login({ identifier: username, password })
+      .then((res) => {
+        if (!res.success) return this.loginState.logout();
+
+        return this.loginState.setLogin(true, res.data.handle);
+      })
+      .catch((error) => {
+        this.logger.error(error);
+        return this.loginState.logout();
+      });
+  }
+
+  createFileModel(): BlueskyFileSubmission {
+    return new BlueskyFileSubmission();
+  }
+
+  calculateImageResize(file: ISubmissionFile): ImageResizeProps {
+    return {
+      // Yes they are this lame: https://github.com/bluesky-social/social-app/blob/main/src/lib/constants.ts
+      height: 2000,
+      width: 2000,
+    };
+  }
+
+  async onPostFileSubmission(
+    postData: PostData<BlueskyFileSubmission>,
+    files: PostingFile[],
+    batchIndex: number,
+    cancellationToken: CancellableToken,
+  ): Promise<PostResponse> {
+    cancellationToken.throwIfCancelled();
+
+    const agent = this.createAgent();
```

Comment:
suggestion: store the agent at login time, instead of recreating for posting unless there is concern about some kind of token timeout.